cpu/stm32/periph/adc_f0_g0.c

```diff
@@ -75,6 +75,19 @@ int adc_init(adc_t line)
     }
     /* reset configuration */
     ADC1->CFGR2 = 0;
+    ADC1->CR |= ADC_CR_ADVREGEN;
+    ADC1->CR &= ~ADC_CR_ADCAL;
+    // calibrate ADC, per RM0454 section 14.3.3
+    // 1. ensure ADEN=0, ADVREGEN=1, DMAEN=0
+    if (!(ADC1->CR & ADC_CR_ADEN) && (ADC1->CR & ADC_CR_ADVREGEN) && !(ADC1->CFGR1 & ADC_CFGR1_DMAEN))
```

Comment:
Under which circumstances would those not be set?
You are already setting `ADVREGEN` in the line above, so why not also

```C
ADC1->CR &= ~(ADC_CR_ADCAL | ADC_CR_ADEN | ADC_CFGR1_DMAEN);
```

Reply:
I think with 'ensure' they mean you should set those bits